generate_compile_commands.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+INSTALL_DIR="/usr/local/bin"
+VERSION="0.5.2"
+
+# Download and symlink.
+(
+  cd "${INSTALL_DIR}" \
+  && curl -L "https://github.com/grailbio/bazel-compilation-database/archive/${VERSION}.tar.gz" | tar -xz \
+  && ln -f -s "${INSTALL_DIR}/bazel-compilation-database-${VERSION}/generate.py" bazel-compdb
+)
+
+bazel-compdb # This will generate compile_commands.json in your workspace root.
```

Comment:
This is a rather aggressive way of installing and handling a file system. It works fine on a disposable build action machine but it is rather aggressive if a developer wanted to run this locally.

May I suggest a modification where a check is made for if `bazel-compdb` is already present before trying to dowload/install:

```bash
# Check if `bazel-compdb` is already present
which bazel-compdb &> /dev/null || (
 # Download and install
)
```

Id further suggest that the `tar` is downloaded to `/tmp/` and that the `tar` is unpacked to `/opt/` (i.e. `INSTALL_PATH` is in `/opt`). This will also affect the symlinking but that should be a detail. :)

Reply:
Totally agree with @while-false, let's not install anything on the developers filesystem outside of the workspace!

Let's instead download into the `dev-tools` directory and add it to `.gitignore` and just call the Python script directly, or is there some reason we need a locally named `bazel-compdb`?

And are we guaranteed that the resulting file will end up in the workspace root? Or do we have to call this from the workspace root? If we need to call this from the workspace root then it would be better to have this script change to the workspace root and then invoke the Python script.